File: frontend/src/types/patient.ts
export type TriageLevel = 'CRITICAL' | 'URGENT' | 'STANDARD';

export interface Patient {
  id: string; // Unified to string for MongoDB compatibility
  name: string;
  age?: number;
  gender?: string;
  chiefComplaint?: string;
  triageLevel: TriageLevel;
  status?: string;
  zoneName?: string;
  roomCode?: string;
  assignedNurse?: string;
  assignedDoctor?: string;
  symptoms?: string[];
  vitals?: {
    heartRate?: number;
    bloodPressure?: string;
    temperature?: number;
    oxygenSaturation?: number;
  };
  createdAt?: string;
  updatedAt?: string;
}

export interface Zone {
  id: string;
  name: string;
  severityBand: TriageLevel;
  description: string;
}

export interface Room {
  id: string;
  roomCode: string;
  zoneId: string;
  zoneName: string;
  equipment: string[];
  occupied: boolean;
  patientId?: string;
}

File: backend/src/main/java/com/mediscan/model/Patient.java
package com.mediscan.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Document(collection = "patients")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Patient {

    @Id
    private String id;

    private String name;
    private Integer age;
    private String gender;
    private String email;
    private String phoneNumber;
    private String chiefComplaint;
    private String symptoms;
    private String vitalSigns;
    private Integer painLevel;
    private String triageLevel;
    private String status;
    private String zoneName;
    private String roomCode;
    private String assignedNurse;
    private String assignedDoctor;
    private LocalDateTime registrationDate;
    private LocalDateTime lastUpdated;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

}
